cmd/rpcdaemon/commands/parity_api.go

```diff
@@ -0,0 +1,75 @@
+package commands
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+
+	"github.com/ledgerwatch/erigon-lib/kv"
+	"github.com/ledgerwatch/erigon/common"
+	"github.com/ledgerwatch/erigon/common/hexutil"
+	"github.com/ledgerwatch/erigon/core/state"
+)
+
+// ParityAPI the interface for the parity_ RPC commands
+type ParityAPI interface {
+	ListStorageKeys(ctx context.Context, account common.Address, quantity int, offset *hexutil.Bytes) ([]hexutil.Bytes, error)
+}
+
+// ParityAPIImpl data structure to store things needed for parity_ commands
+type ParityAPIImpl struct {
+	db kv.RoDB
+}
+
+// NewParityAPIImpl returns ParityAPIImpl instance
+func NewParityAPIImpl(db kv.RoDB) *ParityAPIImpl {
+	return &ParityAPIImpl{
+		db: db,
+	}
+}
+
+// ListStorageKeys implements parity_listStorageKeys. Returns all storage keys of the given address
+func (api *ParityAPIImpl) ListStorageKeys(ctx context.Context, account common.Address, quantity int, offset *hexutil.Bytes) ([]hexutil.Bytes, error) {
+	tx, txErr := api.db.BeginRo(ctx)
+	if txErr != nil {
+		return nil, fmt.Errorf("listStorageKeys cannot open tx: %w", txErr)
+	}
+	defer tx.Rollback()
+	a, err := state.NewPlainStateReader(tx).ReadAccountData(account)
```

Comment:
Get incarnation here, and add it to Seek key

Reply:
added